clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1652,6 +1652,16 @@ void Sema::ConstructOpenCLKernel(FunctionDecl *KernelCallerFunc,
       constructKernelName(*this, KernelCallerFunc, MC);
   StringRef KernelName(getLangOpts().SYCLUnnamedLambda ? StableName
                                                        : CalculatedName);
+  if (KernelLambda->isLambda()) {
```

Comment:
BTW, asking `isLambda` from `KernelLambda` is funny :), we have to rename it in some separate change since it can be also a function object.

Reply:
Addressed this here https://github.com/intel/llvm/pull/2031 .